### PR TITLE
Password confirmation testing

### LINKED ISSUE #3: Write test to confirm password confirmation works as expected
In User New Spec:
- Fill out all fields correctly except password_confirmation
- Expect path to be /register, and display flash notice

### PR DESCRIPTION
**Pull request by nmking22**
This PR will:
* Add functionality for non-matching password and password_confirmation fields in new user form to display a unique message
* Add testing for this functionality

Resolves #3

### CODE REVIEW

File: spec/features/users/new_spec.rb [hunk 1]
```diff
@@ -71,4 +72,23 @@
...
+    fill_in :state,  with: "Texas"
+    fill_in :zip, with: "11111"
+    fill_in :email, with: "shakenbake@gmail.com"
+    fill_in :password, with: "password"
+    fill_in :password_confirmation, with: "password123"
+
+    click_on "Create Account"
+
+    expect(current_path).to eq("/register")
+    expect(page).to have_content("Password and Password Confirmation fields did not match.")
```
Comment (helloeduardo) [inline spec/features/users/new_spec.rb:91]: Thanks for adding this in!

**Review by helloeduardo — APPROVED** [review]
Nice addition to our tests

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -11,7 +11,10 @@ def create
       flash[:notice] = "You are now registered and logged in!"
       redirect_to '/profile'
     elsif @user.duplicate_email?
-      flash[:notice] = "Email already exists, please choose a different email."
```
Comment (the-color-bliu) [inline app/controllers/users_controller.rb:14]: Good catch with the flash :notice vs :failure

**Comment by helloeduardo** [comment]
Also, I think GitHub wants you to use the magic words, resolves, fixes, or closes right before the hashtag and the issue number like this: closes #3 in order to properly link the issue! I don't think closes issue #3 works quite the same way. The more you know!

**Review by the-color-bliu — APPROVED** [review]
Nice job being robust and putting this test in here

**Comment by nmking22** [comment]
> Also, I think GitHub wants you to use the magic words, resolves, fixes, or closes right before the hashtag and the issue number like this: closes #3 in order to properly link the issue! I don't think closes issue #3 works quite the same way. The more you know!

Good to know! I'll edit this on my open PRs and hopefully it will work!

**Pull request merged**